RootOfTrust/RootOfTrust.cs

```diff
@@ -56,13 +58,50 @@ public static bool RegisterIssuer(string issuerDID, string schemaName)
         /// <param name="schemaName">schemaName (string)</param>
         public static bool DeactivateIssuer(string issuerDID, string schemaName)
         {
-            if (!Runtime.CheckWitness(Owner)) throw new Exception("No authorization.");
-            StorageMap issuerTrustList = new StorageMap(Storage.CurrentContext, issuerDID);
+            var owner = GetOwner();
+            if (!Runtime.CheckWitness(owner)) throw new Exception("No authorization");
+
+            StorageMap issuerTrustList = GetIssuerTrustList(issuerDID);
             if (issuerTrustList[schemaName] is null) throw new Exception("No such issuer-schema pair registered");
 
             issuerTrustList.Delete(schemaName);
 
             return true;
         }
+
+        [DisplayName("_deploy")]
+        public static void OnDeploy(object data, bool update)
+        {
+            if (update) return;
+
+            var tx = (Transaction)Runtime.ScriptContainer;
+            var key = new byte[] { Prefix_Owner };
+            Storage.Put(Storage.CurrentContext, key, tx.Sender);
+        }
+
+        public static void Update(ByteString nefFile, string manifest)
+        {
+            var owner = GetOwner();
+            var tx = (Transaction)Runtime.ScriptContainer;
+
+            if (!owner.Equals(tx.Sender) || !Runtime.CheckWitness(owner))
+            {
+                throw new Exception("Only contract owner can update the contract");
+            }
+
+            ContractManagement.Update(nefFile, manifest, null);
+        }
+
+        static StorageMap GetIssuerTrustList(string issuerDID) 
+        {
+            var prefix = (new byte[] { Prefix_Issuers }).Concat(issuerDID);
```

Comment:
I had to add the prefix to the issuerDID to ensure there could not be a collision with the owner prefix.